megamek/src/megamek/utils/BoardsTagger.java

```diff
@@ -0,0 +1,286 @@
+                foliage += hex.containsTerrain(FOLIAGE_ELEV, 1) ? 1 : 0;
+                roads += hex.containsAnyTerrainOf(ROAD, BRIDGE) ? 1 : 0;
+                roughs += hex.containsTerrain(ROUGH) ? 1 : 0;
+                swamps += hex.containsTerrain(SWAMP) ? 1 : 0;
+                lavas += hex.containsTerrain(MAGMA) ? 1 : 0;
+                snowTerrain += hex.containsTerrain(SNOW) ? 1 : 0;
+                ice += hex.containsTerrain(ICE) ? 1 : 0;
+                cliffsTO += hex.containsTerrain(CLIFF_TOP) ? 1 : 0;
+                highCliffs += hex.containsTerrain(INCLINE_HIGH_TOP) ? 1 : 0;
+                fields += hex.containsTerrain(FIELDS) ? 1 : 0;
+                if (hex.getTheme() != null && !hex.containsTerrain(WATER)) {
+                    deserts += hex.getTheme().equalsIgnoreCase("desert") ? 1 : 0;
+                    lunar += hex.getTheme().equalsIgnoreCase("lunar") ? 1 : 0;
+                    grass += hex.getTheme().equalsIgnoreCase("grass") ? 1 : 0;
+                    grass += hex.getTheme().equalsIgnoreCase("") ? 1 : 0;
+                    tropical += hex.getTheme().equalsIgnoreCase("tropical") ? 1 : 0;
+                    mars += hex.getTheme().equalsIgnoreCase("mars") ? 1 : 0;
+                    snowTheme += hex.getTheme().equalsIgnoreCase("snow") ? 1 : 0;
+                    volcanic += hex.getTheme().equalsIgnoreCase("volcanic") ? 1 : 0;
+                } else {
+                    grass++;
+                }
+                wEdgeWater += ((x == 0) && hex.containsTerrain(WATER)) ? 1 : 0;
+                eEdgeWater += ((x == board.getWidth() - 1) && hex.containsTerrain(WATER)) ? 1 : 0;
+                nEdgeWater += ((y == 0) && hex.containsTerrain(WATER)) ? 1 : 0;
+                sEdgeWater += ((y == board.getHeight() - 1) && hex.containsTerrain(WATER)) ? 1 : 0;
+                weighedLevels += Math.abs(hex.getLevel());
+                water += hex.containsTerrain(WATER) ? 1 : 0;
+                if (hex.containsTerrain(BUILDING) 
+                        && (!hex.containsTerrain(BLDG_CLASS) || hex.terrainLevel(BLDG_CLASS) == Building.STANDARD)
+                        && ((hex.terrainLevel(BUILDING) == Building.LIGHT) || (hex.terrainLevel(BUILDING) == Building.MEDIUM))) {
+                    stdBuildings++;
+                    int height = hex.terrainLevel(BLDG_ELEV); 
+                    lowBuildings += (height <= 2) ? 1 : 0;
+                    highBuildings += (height > 2) ? 1 : 0;
+                }
+            }
+        }
+
+        // The board area
+        int area = board.getWidth() * board.getHeight();
+        // The geometric mean of the board sides. Better for some comparisons than the area.
+        int normSide = (int) Math.sqrt(area);
+        int levelExtent = board.getMaxElevation() - board.getMinElevation();
+
+        // Calculate which tags apply
+        Tags.TAG_MEDFOREST.applies = (forest >= normSide * 5) && (forest < normSide * 10) && (forestHU < normSide * 2);

```

Comment:
This is odd and limits it's usage. Let's find another way, perhaps a map.

Reply:
What do you mean? 
After some thinking and deleting answers I wrote my conclusion is you mean: Map<Tags, Boolean> to avoid the "applies" field that is without any sense outside of the BoardTagger. Correct? OK I can see that.

Reply:
Yes, also not thread safe and wouldn't be usable from testing if tested in parallel.